File: Batch Wise/MERN start - 24-2-24/JS/for_loop.js
//every loop must follow 3 rules
//1. initialization
//2. condition to terminate
//3. inc/dec
//for loop syntax
//for(initialization;condition;inc/dec)

//print 1-5 numbers using for loop
for(let i=1; i<=5; i++)
   {
      console.log(i);
   }

// 1 4 27 256 3125
for(let i=1; i<=5; i++)
   {
      console.log(i**i);
   }

// for loop on array
console.log("-- for loop on array type 1 --");
prices = [199,99,49,199,299,49]
for(let i=0; i<prices.length; i++)
   {
      console.log(prices[i]);
   }

console.log("-- for loop on array type 2 --");
for (price of prices)
   {
      console.log(price);
   }

console.log("-- for loop on array type 3 --");
for (price in prices)
   {
         console.log(price,prices[price]);
   }

File: Batch Wise/JAVA 2303/Flow Control/loops/ForLoop.java
public class ForLoop {
   public static void main(String[] args) {
      //initialization - condition - inc/dec - statements (code/body)
      //for(initialization ; condition ; inc/dec) {statements}
      //print 1 to 5 numbers
      System.out.println("print 1 to 5 numbers");
      for(int i=1; i<=5; i++)
      {
         System.out.println(i);
      }
      //print 5 to 1 numbers
      System.out.println("print 5 to 1 numbers");
      for(int i=5; i>=1; i--)
      {
         System.out.println(i);
      }

      //for loop on string and array
      System.out.println("for loop on string");
      String name = "amit";
      for (int i=0; i<name.length(); i++)
      {
         System.out.println(name.charAt(i));
      }

      System.out.println("for loop on array");
      int marks[] = {95,96,93,99};
      for(int i=0; i<marks.length; i++)
      {
         System.out.println(marks[i]);
      }

      System.out.println("for each loop on array");
      for(int mark:marks)
      {
         System.out.println(mark);
      }

   }
}
